### PR TITLE
Phishing - fix for nested email files + new field (message ID)

### PR DESCRIPTION
**Pull request by idovandijk**
## Status
Ready

## Related Issues
Fixes: https://github.com/demisto/etc/issues/46956 (Reported Email Message ID)
Fixes: https://github.com/demisto/etc/issues/46997 (Nested emails support)
Related to: https://github.com/demisto/etc/issues/45133 (ParseEmailFiles fix)

## Description
* Added the `Reported Email Message ID` field - used to differentiate between the message ID of the **reporter's** email, and the **reported** email. Needed for the [delete email through phishing layout](https://github.com/demisto/etc/issues/46959) issue. The field was added to the pack, to the layout (instead of the regular Message ID field), and `Set` inside the process email playbooks (generic + core)
* Added the `EmailFileToExtract` input to the `Phishing - Generic v3` playbook with a default value of `Inner file` - used to fix the problem of multiple email extractions in phishing. For the core playbook, it's hardcoded to use the Inner file (to keep the playbook simple as it only has 1 input today).

## Minimum version of Cortex XSOAR
6.1.0

## Does it break backward compatibility?
No

### CODE REVIEW

File: Packs/Phishing/Playbooks/Process_Email_-_Core_v2.yml [hunk 1]
```diff
@@ -1157,5 +1167,5 @@ outputs:
   description: The file object.
   type: unknown
 tests:
-- Phishing - Core - Test - Incident Starter
```
Comment (tomer-pan) [inline Packs/Phishing/Playbooks/Process_Email_-_Core_v2.yml:1160]: We should keep the test playbook

File: Packs/Phishing/Playbooks/Process_Email_-_Generic_v2.yml [hunk 2]
```diff
@@ -2279,5 +2288,5 @@ outputs:
   description: The file object.
   type: string
 tests:
-- Process Email - Generic - Test - Incident Starter
```
Comment (tomer-pan) [inline Packs/Phishing/Playbooks/Process_Email_-_Generic_v2.yml:2282]: Playbook test was changed

File: Packs/Phishing/Playbooks/Process_Email_-_Generic_v2.yml [hunk 3]
```diff
@@ -2279,5 +2288,5 @@ outputs:
...
-fromversion: 6.0.0
```
Comment (tomer-pan) [inline Packs/Phishing/Playbooks/Process_Email_-_Generic_v2.yml:2283]: We should keep 6.0.0

File: Packs/Phishing/Playbooks/Process_Email_-_Core_v2.yml [hunk 4]
```diff
@@ -1157,5 +1167,5 @@ outputs:
...
-fromversion: 6.0.0
```
Comment (tomer-pan) [inline Packs/Phishing/Playbooks/Process_Email_-_Core_v2.yml:1161]: We should keep 6.0.0

**Comment by idovandijk** [comment]
@jrosenpanw can you please review the following parts:

https://github.com/demisto/content/blob/333aef6fe719ba04222e6439201136cb778adbe3/Packs/Phishing/IncidentFields/incidentfield-Reported_Email_Message_ID.json
line 7

https://github.com/demisto/content/blob/333aef6fe719ba04222e6439201136cb778adbe3/Packs/Phishing/Playbooks/Phishing_-_Generic_v3.yml
lines 3063-3079
**same as** https://github.com/demisto/content/blob/449afcaf840e0cf439d1429a03704da6a07d5dcc/Packs/Phishing/Playbooks/Process_Email_-_Generic_v2.yml
lines 2264-2274


https://github.com/demisto/content/blob/333aef6fe719ba04222e6439201136cb778adbe3/Packs/Phishing/ReleaseNotes/3_1_3.md
release notes

File: Packs/Phishing/Playbooks/Process_Email_-_Core_v2.yml (hunk 4 above)
Comment (idovandijk) [reply Packs/Phishing/Playbooks/Process_Email_-_Core_v2.yml]: Fixed

File: Packs/Phishing/Playbooks/Process_Email_-_Generic_v2.yml (hunk 3 above)
Comment (idovandijk) [reply Packs/Phishing/Playbooks/Process_Email_-_Generic_v2.yml:2283]: Fixed

File: Packs/Phishing/Playbooks/Process_Email_-_Core_v2.yml (hunk 1 above)
Comment (idovandijk) [reply Packs/Phishing/Playbooks/Process_Email_-_Core_v2.yml]: Fixed

**Review by tomer-pan — APPROVED** [review]

**Comment by tschanfeld** [comment]
@idovandijk Made minor language edits to requested files

**Pull request merged**